### ISSUE TITLE
SEGV whenever other outputs change 

### ISSUE THREAD

**Issue by colemickens** [any1/wayvnc#45]
1. I was VNC'd into my laptop running Sway. At this time, the KVM was not on the laptop, so the only output was HEADLESS-1.
2. The KVM switched back to the laptop, enabling the other output.

At this point my VNC session crashed. Sway kept running and I was able to just restart wayvnc, but it would be nice if it didn't crash of course.

```
           PID: 28857 (wayvnc)
           UID: 1000 (cole)
           GID: 1000 (cole)
        Signal: 11 (SEGV)
     Timestamp: Wed 2020-07-01 16:00:06 PDT (10min ago)
  Command Line: wayvnc -o HEADLESS-1 0.0.0.0
    Executable: /nix/store/ravgkjx5c1bssi4svnnbh05cz1p05508-wayvnc-4f1bca55c3bf19a58e9d892a5141fdfc5159bf01/bin/wayvnc
 Control Group: /user.slice/user-1000.slice/session-8.scope
          Unit: session-8.scope
         Slice: user-1000.slice
       Session: 8
     Owner UID: 1000 (cole)
       Boot ID: 7afde54d612c4eaebc38d2961d43f7ed
    Machine ID: befa945b3fb44f73a9671674ee23e039
      Hostname: xeep
       Storage: /var/lib/systemd/coredump/core.wayvnc.1000.7afde54d612c4eaebc38d2961d43f7ed.28857.1593644406000000000000.lz4
       Message: Process 28857 (wayvnc) of user 1000 dumped core.

Reading symbols from /nix/store/ravgkjx5c1bssi4svnnbh05cz1p05508-wayvnc-4f1bca55c3bf19a58e9d892a5141fdfc5159bf01/bin/wayvnc...
(No debugging symbols found in /nix/store/ravgkjx5c1bssi4svnnbh05cz1p05508-wayvnc-4f1bca55c3bf19a58e9d892a5141fdfc5159bf01/bin/wayvnc)
[New LWP 28857]
[New LWP 28864]
[New LWP 28858]
[New LWP 28859]
[New LWP 28860]
[New LWP 28861]
[New LWP 28867]
[New LWP 28869]
[New LWP 28865]
[New LWP 28866]
[New LWP 28868]
[New LWP 28863]
[New LWP 28870]
[New LWP 28862]

warning: Unexpected size of section `.reg-xstate/28857' in core file.
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/nix/store/bqbg6hb2jsl3kvf6jgmgfdqy06fpjrrn-glibc-2.30/lib/libthread_db.so.1".
Core was generated by `wayvnc -o HEADLESS-1 0.0.0.0'.
Program terminated with signal SIGSEGV, Segmentation fault.

warning: Unexpected size of section `.reg-xstate/28857' in core file.
#0  0x00007f90f2e44209 in wl_proxy_marshal () from /nix/store/qk4lyaah441wxakif33im54ww25lsb7d-wayland-1.18.0/lib/libwayland-client.so.0
[Current thread is 1 (Thread 0x7f90f24a3ac0 (LWP 28857))]
(gdb) bt
#0  0x00007f90f2e44209 in wl_proxy_marshal () from /nix/store/qk4lyaah441wxakif33im54ww25lsb7d-wayland-1.18.0/lib/libwayland-client.so.0
#1  0x00000000004085da in output_destroy ()
#2  0x0000000000405bce in registry_remove ()
#3  0x00007f90f2b44abd in ffi_call_unix64 () from /nix/store/gbwpr7mphpk1fyfz1dn05n2i00aspwfl-libffi-3.3/lib/../lib64/libffi.so.7
#4  0x00007f90f2b4379c in ffi_call_int () from /nix/store/gbwpr7mphpk1fyfz1dn05n2i00aspwfl-libffi-3.3/lib/../lib64/libffi.so.7
#5  0x00007f90f2e471b0 in wl_closure_invoke () from /nix/store/qk4lyaah441wxakif33im54ww25lsb7d-wayland-1.18.0/lib/libwayland-client.so.0
#6  0x00007f90f2e43ad8 in dispatch_event.isra () from /nix/store/qk4lyaah441wxakif33im54ww25lsb7d-wayland-1.18.0/lib/libwayland-client.so.0
#7  0x00007f90f2e4508c in wl_display_dispatch_queue_pending () from /nix/store/qk4lyaah441wxakif33im54ww25lsb7d-wayland-1.18.0/lib/libwayland-client.so.0
#8  0x0000000000405af8 in on_wayland_event ()
#9  0x00007f90f2e7c41a in aml_dispatch () from /nix/store/apqhrq1lkz67pqqig15p6bhb8k3cwgkx-aml-099e0c71942a0c0a42f40b9b18e616cfe20dd534/lib/libaml.so.0
#10 0x0000000000405097 in main ()
```

**Comment by any1** [comment]
Does this still happen?

**Comment by helloworld1** [comment]
Still happens to me. Segfault when output changes
wayvnc -V
wayvnc: 0.3.1
neatvnc: 0.3.2
aml: 0.1.0

**Comment by lack** [comment]
Pretty sure this is solved by #201 - Please reopen if not.
